cap-containers/map.h

```diff
@@ -165,6 +174,25 @@ static cap_map *cap_map_init(size_t key_size,
 	map->_key_size = key_size;
 	return map;
 }
+static cap_map_iterator *cap_map_iterator_init(cap_map *map){
+	assert(map != NULL);
+	cap_map_iterator *iterator = 
+	    (cap_map_iterator *)CAP_ALLOCATOR(cap_map_iterator, 1);
+	if (!iterator) {
+		fprintf(stderr, "memory allocation failue\n");
+		return NULL;
+	}
+	if (!map){
+		iterator->_current_index = 0;
+		map->_key[0]=NULL;
+		map->_value=NULL;
```

Comment:
It looks like you're not using the `clang-format`.

Our code base has a specific coding style which is defined on the `.clang-format` file at the root of the project. It defines rules for the spacing, tabs, line wrap, and all other such Infos.

If you don't have `clang-format`, You can install it via `apt-get install clang-format`(assuming you're on Ubuntu)

Before committing and pushing into the remote repo, please goto the root of the project and run the util 
`clang-format` to format the code which matches the coding style.

Ex:
After making changes, do the following before committing and pushing.
```bash
$ clang-format -i path_to_source_file_you_edited.h
```

This will make sure to format the code for you, as per the rules defined in `.clang-format`